pkg/events/events.go

```diff
@@ -21,6 +21,11 @@ type probeDependency struct {
 	Required bool // should tracee fail if probe fails to attach
 }
 
+type kSymbolsDependencies struct {
+	RequireDynamicSymbols bool
+	StaticSymbols         []kSymbolDependency
```

Comment:
They're not, for example `hooked_proc_fops` needs both.
And if we go by the nil suggestion then all events will have a dynamic symbols requirement, which isn't true. As such tracee's memory footprint will include the ksymbols even if no ksymbols related events are traced.

Reply:
> They're not, for example hooked_proc_fops needs both.

RequireDynamicSymbols is set to true only in the new mem_dump event, and not for hooked_proc_fops.

> And if we go by the nil suggestion then all events will have a dynamic symbols requirement, which isn't true. As such tracee's memory footprint will include the ksymbols even if no ksymbols related events are traced.

I suggested differentiating between three options:
1. No dependencis -> nil (then we should NOT ask for dynamic symbols)
2. Static dependencies -> list is not empty, ask only for those symbol dependencies
3. Dynamic dependencies -> list is empty -> set the dependency

Reply:
> RequireDynamicSymbols is set to true only in the new mem_dump event, and not for hooked_proc_fops.
It should have both but it's practically not needed because by setting the static symbols the kernel symbols object is loaded and dynamic querying is enabled.

> I suggested differentiating between three options:
You're right, i've misread what you wrote then, that can work as well.